### PR TITLE
metadata-writer: Fix multi-user mode

### PR DESCRIPTION
**Pull request by yanniszark**
Fixes https://github.com/kubeflow/pipelines/issues/5413

The metadata-writer should use a different function to watch Pods, if
the namespace is empty, i.e, if it wants to watch Pods in all
namespaces.

/cc @Bobgy

### CODE REVIEW

**Comment by Bobgy** [comment]
/cc @Ark-kun 
Thank you for the fix!

It was working properly before, did anything change?

File: backend/metadata_writer/src/metadata_writer.py [hunk 1]
```diff
@@ -137,13 +137,21 @@ def is_kfp_v2_pod(pod) -> bool:
...
-    for event in k8s_watch.stream(
+    pod_stream = k8s_watch.stream(
         k8s_api.list_namespaced_pod,
         namespace=namespace_to_watch,
         label_selector=ARGO_WORKFLOW_LABEL_KEY,
         timeout_seconds=1800,  # Sometimes watch gets stuck
         _request_timeout=2000,  # Sometimes HTTP GET gets stuck
-    ):
+    )
+    if not namespace_to_watch:
```
Comment (Bobgy) [inline backend/metadata_writer/src/metadata_writer.py:147]: the first watch is still called when namespace_to_watch is empty

Can there be side effects? Shall we put both branches into an if else?

Comment (yanniszark) [reply backend/metadata_writer/src/metadata_writer.py:147]: You are correct! I will push a fix immediately.

File: backend/metadata_writer/src/metadata_writer.py [hunk 2]
```diff
@@ -137,13 +137,22 @@ def is_kfp_v2_pod(pod) -> bool:
...
     print("Start watching Kubernetes Pods created by Argo")
-    for event in k8s_watch.stream(
-        k8s_api.list_namespaced_pod,
-        namespace=namespace_to_watch,
-        label_selector=ARGO_WORKFLOW_LABEL_KEY,
-        timeout_seconds=1800,  # Sometimes watch gets stuck
-        _request_timeout=2000,  # Sometimes HTTP GET gets stuck
-    ):
+    if namespace_to_watch:
+        pod_stream = k8s_watch.stream(
```
Comment (Bobgy) [inline backend/metadata_writer/src/metadata_writer.py:141]: I think in python scope of this var is now limited to this if branch, you need to declare

pod_stream = None

before the if command

Can you verify?

**Comment by Bobgy** [comment]
/lgtm
/approve
You are right! Thank you for the update!

**Pull request merged**